File: src/hexmap/TacticalHexCell.ts
import { WeatherType } from './HexCell';

export interface TacticalHexCell {
  q: number;
  r: number;
  terrainType: string;
  terrainEffect: string;
  cover: number;
  movementCost: number;
  visibility: number;
  elevation?: number;
  features?: string[];
  environmentalEffects?: Array<{
    type: string;
    magnitude: number;
    duration?: number;
  }>;
} 

File: backend/app/src/hexmap/TacticalHexCell.py
from typing import Any, List


class TacticalHexCell:
    q: float
    r: float
    terrainType: str
    terrainEffect: str
    cover: float
    movementCost: float
    visibility: float
    elevation?: float
    features?: List[str]
    environmentalEffects?: List[{
    type: str
    magnitude: float
    duration?: float>
} 